src/realm/list.hpp

```diff
@@ -68,6 +68,7 @@ class LstBase : public CollectionBase {
     virtual void swap(size_t ndx1, size_t ndx2) = 0;
 
 protected:
+    static constexpr CollectionType s_collection_type = CollectionType::List;
```

Comment:
Why not to have a CollectionBase::collection_type variable and let the implementor set that?

Reply:
If we can replace a variable with a constant, that is better.